Query and Update animals table

harshi0102 opened the pull request:
In this pull request:
•	I added a column species of type string to your animal's table and Modified my schema.sql file using ALTER command.
•	I inserted the following data into the data.sql file:

- Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has
never tried to escape.
- Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has
tried to escape 2 times.
- Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he
has tried to escape 3 times.
- Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried
to escape once.
- Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried
to escape 7 times.
- Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried
to escape 3 times.
-Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to
escape 4 times.

![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/9ba88327-b3c3-44c3-9b67-6ea29bc55561)

**•	I added a transaction update for the animals' table by setting the species to unspecified and rollback.**
![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/58dc99b3-a67d-4a80-a854-b3f0a04f32de)


•	**I updated the animals' table by setting the species column to digimon for all animals that have a name ending in mon.
•	I updated the animals' table by setting the species column to pokemon for all animals that don't have species already set.
•	I committed the transaction and verified the changes made persisted after the commit.**

![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/e43e8570-6a7f-4b58-9e9e-a34f5ea8c026)

**•	In a delete transaction, all the animals in the table were deleted. Then I rolled back the transaction. After the rollback, I verified if all the transactions exists.**
![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/e8c3f902-509f-4af3-88e8-c82cb28b8bb3)

**•	In the next transaction I deleted all animals born on January 1st, 2022, then created a savepoint for the transaction.**
![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/67b48315-c5cb-4109-a4e0-69328ce09a2c)

**•	I updated the animals' weight to be their weight multiplied by -1 then rolled back the savepoint.**
![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/4b7fe84d-7905-4f9b-946c-af9a2ca38dc2)

**•	I updated all animals' weights that are negative to be their weight multiplied by -1 and then committed the transaction.**
![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/941b5e59-0971-485c-83ed-a4826ebe8d3b)

**Write queries to answer the following questions:**
 **How many animals are there?**
 
![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/b40703fb-8d89-46ee-8f18-fbe84c6807d4)


 **How many animals have never tried to escape?**
 
![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/a1badef0-289a-42cb-8225-3dfa96887078)


**What is the average weight of animals?**
![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/5420189b-65ad-458c-90ce-6ba3561cbe0e)


**Who escapes the most, neutered or not neutered animals?**
 
![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/d777e637-1859-4452-850d-dda640dd4167)

**What is the minimum and maximum weight of each type of animal?**
 
![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/7db643d1-ab7e-4261-acf7-529134ffa5e9)


**What is the average number of escape attempts per animal type of those born between 1990 and 2000?**
 
![image](https://github.com/harshi0102/vet-clinic-project/assets/108334376/26e19bbb-2ee7-4579-ad60-2259987be09c)

tresorsawasawa commented on queries.sql:
```diff
@@ -1,11 +1,60 @@
+ROLLBACK TO SAVEPOINT_1;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+
```
- [ ] `README.md` file: I think it would be better if you fix the broken `[MIT](./LICENSE)` link in your README file to make it more professional as it shows a **404** error page once clicked. And to fix that, you can only create a new file name `LICENSE` in your root directory and get its content from this **[template](https://github.com/microverseinc/readme-template/blob/master/MIT.md)** or rename this `[MIT](./LICENSE)` to `[MIT](./MIT.md)`, then the created file in your directory name it `MIT.md` and use the same comment in this **[template](https://github.com/microverseinc/readme-template/blob/master/MIT.md)**
![Screenshot from 2023-06-13 18-19-27](https://github.com/harshi0102/vet-clinic-project/assets/83042742/ea27e2af-23fe-4ede-be8e-92ab3e8f6130)

tresorsawasawa commented on queries.sql:
```diff
@@ -1,11 +1,60 @@
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, SUM(escape_attempts) FROM animals GROUP BY neutered;
```
- [ ] Kindly make sure you correctly answer to this query as you are supposed to get the `highest` `escape_attempts` and not the `total` of the `escape_attempts`. If you correctly read the query, it is said: `Who escapes the most, neutered or not neutered animals?`, that means you can use `MAX` instead of `SUM` for example:
```sql
..., MAX(escape_attempts) FROM animals ...
```

tresorsawasawa commented on queries.sql:
```diff
@@ -1,11 +1,60 @@
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, SUM(escape_attempts) FROM animals GROUP BY neutered;
+SELECT species, MAX(weight_kg) AS maximum_weight, MIN(weight_kg) AS minimum_weight FROM animals GROUP BY species;
+SELECT species, ROUND(AVG(escape_attempts), 2) AS average_escape_attempts FROM animals WHERE date_of_birth >= 
+'1990-01-01' AND date_of_birth < '2000-12-31' GROUP BY species;
```
- [ ] For the project requirements and to correctly answer to this query, kindly ensure that animal born on `2000-12-31` are also include as right now they are excluded. It should be something similar to this: 
```sql
...  WHERE date_of_birth >= '1990-01-01' AND date_of_birth <= '2000-12-31' GROUP BY species;
```
Remember to also update your screenshots to ensure that you ran the queries with success.

tresorsawasawa reviewed (changes requested):
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @harshi0102 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

![almost-there](https://user-images.githubusercontent.com/57562869/143511358-9cde7daf-4ad0-4db2-8f15-18b54288ff91.gif)

## To highlight :champagne: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

harshi0102 commented:
According to the above feedback, I added my license link in the readme file and updated the screenshot according to the recent update in the sql queries

ShadyShawkat reviewed (approved):
## Hi @harshi0102, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

![Image](https://media.giphy.com/media/jJQC2puVZpTMO4vUs0/giphy.gif)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
